lib/cli-engine.js

```diff
@@ -372,31 +398,49 @@ CLIEngine.prototype = {
     executeOnFiles: function(files) {
 
         var results = [],
-            processed = [],
+            processed = {},
             options = this.options,
             configHelper = new Config(options),
-            ignoredPaths = IgnoredPaths.load(options),
-            exclude = ignoredPaths.contains.bind(ignoredPaths),
+            ignoredPaths = options.ignore ? IgnoredPaths.load(options).patterns : [],
+            extensions = options.extensions.map(function(ext) {
+                return ext.charAt(0) === "." ? ext : "." + ext;
+            }),
+            globOptions,
             stats;
 
-        traverse({
-            files: files,
-            extensions: options.extensions,
-            exclude: options.ignore ? exclude : false
-        }, function(filename) {
+        files = files.map(processPath);
+        ignoredPaths = ignoredPaths.map(processPath);
+
+        globOptions = {
+            nodir: true,
+            realpath: true,
+            ignore: ignoredPaths
+        };
+
+        function executeOnFile(filename) {
```

Comment:
That spot is in a function that is run in a `forEach` loop. That's why I moved it to the outer scope.